### DISCUSSION TITLE
Give up GitHub / GitHub itself is not FOSS friendly

### DISCUSSION THREAD

**Discussion by return42** [searxng/searxng#4025] [Ideas]
Dear FOSS developers, how are you doing with GitHub, should we perhaps switch to a different git hoster? 

Personally, I'm only here because of the community ... and as I can read on Matrix, I'm not the only one who has doubts about GitHub.

Github itself is not FOSS and I would rather host my projects on real FOSS git hosters like Codeberg, where we already host our [translations](https://translate.codeberg.org/projects/searxng/searxng/).

Switching to another git hoster will involve a lot of work and we will also lose a significant part of the community.

Whether .. and where we will move to one day does not have to be decided today, but I would like to discuss a move with the community here in this thread.

First of all, we should think about what damage it would cause (to the community) if we change the git hoster. In the second step, we should consider what alternatives there are and which one suits us best..

What suggestions do you have, what would you see as advantageous or disadvantageous? What aspects should we take into consideration?  what ...?

Related:

- [Codeberg](https://codeberg.org/) (powered by Forgejo)
- [Forgejo](https://forgejo.org/)
- [GitLab](https://about.gitlab.com/) ([Why gitlab](https://about.gitlab.com/why-gitlab/))
- [Give up GitHub](https://sfconservancy.org/GiveUpGitHub/)

---

A few personal remarks / thoughts

I used to run GitLab (self-hosting) and have also had experience with [gitlab.com ](https://gitlab.com/explore)
  - I find the UX of github or forgejo better than that of GitLab 
  - GitLab feels bloated to me .. self-hosting a gitlab instance requires some effort for operation and maintenance and is too driven by the usual business trends for my taste / to get an impression: https://about.gitlab.com/why-gitlab/

Forgejo is the “successor” to Gitea .. [Why was forgejo created](https://forgejo.org/compare-to-gitea/#why-was-forgejo-created)

With gitlab.com we have to register OpenSource projects in order to have a similar range of functions in the CI as we know it from github.

I would prefer a simple, not too bloated software; not all the features on GitHub are required.  The repository must also be public without logging in to the hoster.

My current favorite git hoster is codeberg.com;

- codeberg.com it is based on forgejo
- Codeberg e.V. is non-profit organization and the community is comprised by FOSS-minded developers
- *"No tracking. No third-party cookies. No profiteering."*
- Hosted in Europe: European jurisdiction has a stricter regulation on data protection than most other locations on planet earth ([GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation))

We will always need some kind of Git-oster / the maintenance and operation of a self-hosted solution is associated with a lot of effort that we cannot afford / a self-hosting solution would also take us away from any community that is on a git hoster.

**Comment by unixfox** [comment]
Who is going to pay for:
- the CI
- hosting the documentation: https://github.com/searxng/searxng/tree/gh-pages
- checking the uptime of the instances list: https://github.com/searxng/searx-instances-uptime
- hosting the artifacts of the helm chart: https://github.com/searxng/searxng-helm-chart/tree/gh-pages

All of these that we take for granted but GitHub is giving them for free.

**Reply by dalf** [reply]
Also https://github.com/searxng/searxng-browser-tests
published here : https://dev.searxng.org/screenshots.html#config=Firefox+1920&test_case=search_general
